nimare/workflows/cbma.py

```diff
@@ -0,0 +1,91 @@
+"""Workflow for running an coordinates-based meta-analysis from a NiMARE database."""
+import logging
+
+from nilearn import reporting
+
+from nimare.correct import FWECorrector
+from nimare.diagnostics import FocusCounter
+from nimare.meta import ALE
+
+LGR = logging.getLogger(__name__)
+
+
+def cbma_workflow(
+    dataset,
+    meta_estimator=ALE(),
+    corrector=FWECorrector(),
+    diagnostics=(FocusCounter(),),
+    output_dir=None,
+):
+    """Compose a coordinate-based meta-analysis workflow.
+
+    .. versionadded:: 0.0.14
+
+    This workflow performs a coordinate-based meta-analysis, multiple comparison correction,
+    and diagnostics analyses on corrected meta-analytic maps.
+
+    Parameters
+    ----------
+    dataset : :obj:`~nimare.dataset.Dataset`
+        Dataset for which to run meta-analyses to generate maps.
+    meta_estimator : :class:`~nimare.base.CBMAEstimator`, optional
+        Meta-analysis estimator. Default is :class:`~nimare.meta.cbma.ale.ALE`.
+    meta_corrector : :class:`~nimare.correct.Corrector`, optional
+        Meta-analysis corrector. Default is :class:`~nimare.correct.FWECorrector`.
+    diagnostics : :obj:`tuple`, optional
+        Tuple of meta-analysis diagnostic classes.
+        Default is :class:`~nimare.diagnostics.FocusCounter`.
+    output_dir : :obj:`str`, optional
+        Output directory in which to save results. If the directory doesn't
+        exist, it will be created. Default is None (the results are not saved).
+
+    Returns
+    -------
+    :obj:`~nimare.results.MetaResult`
+        Results of Estimator and Corrector fitting with cluster and diagnostic tables.
+    """
+    LGR.info("Performing meta-analysis...")
+    results = meta_estimator.fit(dataset)
+
+    LGR.info("Performing correction on meta-analysis...")
+    corr_results = corrector.transform(results)
+
+    LGR.info("Generating cluster tables and performing diagnostics on corrected meta-analyses...")
+    for img_key in corr_results.maps.keys():
+        # Save cluster table only for z maps
+        if img_key.startswith("z_"):
+            cbma_img = corr_results.get_map(img_key)
+            cluster_df = reporting.get_clusters_table(cbma_img, stat_threshold=0)
```

Comment:
Should we set the cluster forming threshold to something like `3.1`?

Reply:
I think it depends on the Corrector settings and the map. For voxel-level corrected and uncorrected maps, I'd say no cluster-forming threshold should be applied. For the cluster-level corrected maps, then the threshold should match the one used for the Corrector.

Reply:
I'm not sure how to access the `ss_thresh` parameter from the Corrector directly. 
Wouldn't that threshold (`stat_threshold=ss_thresh`) produce the same cluster table as `stat_threshold=0` for the cluster-level corrected maps, given that those maps were already thresholded/corrected?

Reply:
You're totally right... in that case, I think I need to break it down by image:
- z (uncorrected): p < 0.001 might be good here
- cFWE corrected z: The threshold would remove the clusters, so p < 0.05 is a good default here.
- vFWE corrected z: Since it's corrected, we probably want p < 0.05, rather than p < 0.001

Reply:
thanks! and for FDR and FWE bonferroni which default thresholds would you suggest?

Reply:
for bonferroni-corrected p-values, I would say 0.05 as well. 

For the fdr corrected p-values, I would use whatever alpha is set to in the class:
https://github.com/neurostuff/NiMARE/blob/ad09e4fe7fedaaf6a64949c4c14164d1a02a9cf8/nimare/correct.py#L340

Reply:
Sound good!
I'm not really sure how the alpha (`q`) is used here: 
https://github.com/neurostuff/PyMARE/blob/604bb4fc33ff74703f0408ccd2031d3802113702/pymare/stats.py#L178